src/catalog/src/durable/impls/persist.rs

```diff
@@ -273,7 +95,7 @@ impl PersistHandle {
             .open(
                 shard_id,
                 Arc::new(VecU8Schema::default()),
```

Comment:
Is there any world in which we want to write this down instead as a two-column Row/SourceData (an int for the tag and a Vec<u8> for the rest)? That would maybe future-proof us for reading this in dataflows (no idea why we'd want to do that. debugging? maintaining indexes over subsets of it?)

Reply:
There may be a world in which we want this. It might be possible to define materialized views over that structure to implement all of the system tables.

I think it's a bit more likely that once the persist-txn stuff lands, we'd want a separate shard for each `StateUpdateKind` variant, where each of those shards are use Row/SourceData.

Maybe for now a two-column Row/SourceData is the way to go. How would I go about implementing that schema?